File: kotlin/src/main/kotlin/Character.kt
class Character {
    private val MAX_HEALTH = 1000
    private var health = MAX_HEALTH
    fun hasHealth(value: Int): Boolean = health == value
    fun hasLevel(): Int = 1
    fun isAlive(): Boolean = health > 0

    fun damage(value: Int) {
        if (value > health) {
            health = 0
            return
        }

        health -= value
    }

    fun heal(value: Int) = when {
        isDead() -> Unit
        isMaxHealthExceeded(value) -> health = MAX_HEALTH
        else -> health += value
    }

    private fun isMaxHealthExceeded(value: Int) = value + health > MAX_HEALTH

    fun isDead(): Boolean = !isAlive()
}

File: php/src/Character.php
<?php declare(strict_types=1);

namespace Kata;

class Character
{
    private int $health;
    private int $level;

    public function __construct(
        int $health = 1000,
        int $level = 1
    ) {
        $this->health = $health;
        $this->level = $level;
    }

    public function damage(Character $other, int $damage): void
    {
        if ($other->health < $damage) {
            $other->health = 0;
            return;
        }
        $other->health -= $damage;
    }

    public function isAlive(): bool
    {
        return $this->health > 0;
    }

    public function heal(Character $other, int $health): void
    {
        $other->health += $health;
    }
}
